src/main/java/org/humancellatlas/ingest/exporter/DefaultExporter.java

```diff
@@ -83,12 +92,21 @@ public void exportBundles(SubmissionEnvelope envelope) {
 
         int partitionSize = 500;
 
+        JSONObject context = new JSONObject();
+        context.put("totalAssayCount", totalCount);
```

Comment:
Are these export jobs for just the Terra exporter?

Reply:
yup, i believe so. Do we need to create exportJobs for manifest generation?

Reply:
I think so. Otherwise, the terra-exporter and manifest-generator are both going to receive messages that refer to this ExportJob.

Maybe we should create separate exporters for the different kinds of exporter we have, and each one should have knowledge about the "context"

Reply:
In this change, it wouldn't create any export job for the manifest generation.

Reply:
If we want an export job to be created for manifest generation, that means we also have to put that logic of setting the export job to Exported in manifest generation as well.

Reply:
I think manifest generation should be considered part of the archiver's ExportJob.

Maybe we should create separate "exporters" in ingest-core for the different kinds of exporter we have, and each one should have knowledge about the destination and "context". For example, rename this to TerraExporter and also create an ArchiveExporter for manifests generation + whatever else it needs. At some point, we should then consider moving TerraExporter and ArchiveExporter into the actual export components.

Reply:
Yeah i agree it can be part of the archiving.

Optimisation idea: I think Archiving and Exporter could share that assay data. If we model it as Assay data as part of Core, and put all the information like the linking information, assay process, list of donors etc, that both archiver and exporter could use it and we're reducing the duplicate graph crawling which happened in both processes.

For this change, i think we're only aiming for MVP which shouldn't touch archiving yet. We should make that change of creating export job for archiving later in a separate PR.

Reply:
I think optimization is a good idea at some point, but I'm most concerned with ensuring consistency. Inconsistency and resulting bugs are a source of technical debt, unoptimized code is only technical debt if current speed isn't acceptable for users.

> For this change, i think we're only aiming for MVP which shouldn't touch archiving yet. We should make that change of creating export job for archiving later in a separate PR.

Yep, I just want us to be aware of where we're introducing sources of inconsistent behaviour. I think we should document the problems we see in our MVP solutions. We might not get a chance to address this for some time, and I can see it all being a source of headaches in the future.